torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -292,27 +290,26 @@ bool Reducer::ddp_graph_static() {
 }
 
 void Reducer::initialize_local_used_map() {
-  const auto replica_count = replicas_.size();
-  const auto variable_count = replicas_[0].size();
+  const auto replica_count = 1;
+  size_t replica_index = 0;
+  const auto variable_count = replicas_.size();
   local_used_maps_.resize(replica_count);
   local_used_maps_dev_.resize(replica_count);
 
-  for (const auto i : c10::irange(replica_count)) {
-    at::TensorOptions options;
-    options = options.dtype(at::kInt);
-
-    // Deliberately don't pin the memory even if local_used_maps_dev_ will
-    // be cuda. See Note [local_used_maps_ -> local_used_maps_dev copying]
-    local_used_maps_[i] =
-        at::zeros({static_cast<long>(variable_count)}, options);
+  at::TensorOptions options;
+  options = options.dtype(at::kInt);
 
-    // This tensor needs to be on the same device as replica because backend
-    // such as NCCL may not support CPU tensors, and hence it might not work
-    // if we always put it on CPU.
-    options = options.device(replicas_[i][0].device());
-    local_used_maps_dev_[i] =
-        at::empty({static_cast<long>(variable_count)}, options);
-  }
+  // Deliberately don't pin the memory even if local_used_maps_dev_ will
+  // be cuda. See Note [local_used_maps_ -> local_used_maps_dev copying]
+  local_used_maps_[replica_index] =
```

Comment:
Probably out of scope for this PR, but in the future `local_used_maps_` can also be refactored to not have to be indexed.